Dockerfile

```diff
@@ -0,0 +1,39 @@
+ARG JAVA_VERSION=16
+ARG JVM_FLAVOR=hotspot
+
+FROM adoptopenjdk:${JAVA_VERSION}-jdk-${JVM_FLAVOR} AS builder
+WORKDIR /builddir
+
+COPY gradlew build.gradle settings.gradle ./
```

Comment:
Why is this all manual instead of adding the entire directory and instead adding known unnecessary directories to `.dockerignore`?

Reply:
Did it this way initially to allow for having 2 separate build steps (one for backend and one for CLI), before putting them in the final image.
This also has a few advantages:
Build is not invalidated at all if `dockerdefaults.yaml` is changed (minor benefit)
Gradle wrapper download is not invalidated if only source changes. I'll see if copying `build.gradle` later will work too (I assume it should), as then only changes to the wrapper itself will invalidate it.

I forgot that I should probably support legacy docker builder, which sends whole build context eagerly to docker daemon, so should add `.dockerignore` anyway.

Reply:
Would prefer we just add the whole directory. CLI won't be part of this image, so it is not a concern.

Reply:
Was mostly meaning the part where the gradle wrapper is not re-downloaded if it's not required to do so.
I guess I might be optimizing it too much for work flows I know and if Docker is only used for deployment from time to time, it shouldn't matter too much.
Just wanted to avoid pointless minutes wasted waiting for it to download and initialize every time :)

Reply:
Yes - please modify to use whole directory as mentioned.

Reply:
Changed